internal/server/v2/observation/calculation_util.go

```diff
@@ -0,0 +1,168 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package observation
+
+import (
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"reflect"
+	"strings"
+
+	pb "github.com/datacommonsorg/mixer/internal/proto"
+)
+
+// The info of a node in the AST tree.
+type NodeData struct {
+	StatVar string
+	Facet   *pb.Facet
+}
+
+type Calculation struct {
```

Comment:
Since this does not do calculation, maybe call it "VariableFormula"?

Reply:
Done